### PR TITLE
Keepalive - Remove race condition when removing destination

### PR DESCRIPTION
**Pull request by NGSegovia**
#### Pre-Submission Checklist

- [x] Commit message has the format required by CONTRIBUTING guide
- [x] Commits are split per component (core, individual modules, libs, utils, ...)
- [x] Each component has a single commit (if not, squash them into one commit)
- [x] No commits to README files for modules (changes must be done to docbook files
in `doc/` subfolder, the README file is autogenerated)

#### Type Of Change
- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)

#### Checklist:

- [ ] PR should be backported to stable branches
- [x] Tested changes locally
- [x] Related to issue #2448  (replace XXXX with an open issue number)

### CODE REVIEW

File: src/modules/keepalive/keepalive_api.c
Comment (linuxmaniac) [inline src/modules/keepalive/keepalive_api.c:207]: You can't call free_destination(target) and after that use target

**Review by linuxmaniac — CHANGES_REQUESTED** [review]

File: src/modules/keepalive/keepalive_api.c
Comment (NGSegovia) [reply src/modules/keepalive/keepalive_api.c]: Opps, thanks for the catch @linuxmaniac

Comment (NGSegovia) [reply src/modules/keepalive/keepalive_api.c]: Fixed

**Comment by miconda** [comment]
I am going to merge this PR if no other comments. No deep review done by me, I am not using the module.

@NGSegovia - as a matter of fact, give that you seem to work closely with this module, we can grant you git commit access if you want to take over the maintenance of this module. The original developer of the module is no longer working in VoIP space related to it. Even with git commit access, you should consider using the pull requests process for major changes to the module, other devs may have useful feedback before merging.

On another plan, not being really familiar with the code of the module, but based on what I could figure out so far, as general remarks, the keepalive module seems to be designed for dealing with rather small number of destinations, keeping a single linked list. Moreover, it uses the main core time which can affect performance and impact retransmissions of routing SIP requests if it is overloaded by large number of keepalive tasks or the event callbacks executed by keepalive are slow. If someone plans to use it for large number of addresses, it should seriously consider redesigning some internal parts of this module to be more scalable (e.g., use hash table instead of single list, own timer processes, etc...).

**Comment by NGSegovia** [comment]
Thanks @miconda , that would be a pleasure. We are using this module so I think I can take it. 

About the number of destinations, I had the same impression but I was surprised by some feedback in this comment https://github.com/kamailio/kamailio/issues/2448#issuecomment-689451262. Anyway, I still think this module has limitations in that regard.

**Comment by miconda** [comment]
@NGSegovia - developer invitation sent, you can merge at your convenience, provided no other comments are done.

**Comment by NGSegovia** [comment]
Received @miconda! Thanks!

**Pull request merged**
